mbuild/formats/gomc_conf_writer.py

```diff
@@ -0,0 +1,4065 @@
+        Parameters
+        ----------
+        charmm_object :  Charmm object, which by definition has been parameterized ' \
+                      + 'from the selected force field.',
+        ensemble_typ : str, only accepts 'NVT', 'NPT', 'GEMC_NPT', 'GCMC-NVT', 'GCMC'
+        RunSteps : int; must be an integer greater than zero.
+            Sets the total number of simulation steps.
+        Temperature : Temperature of system in Kelvin (K)
+
+        input_variables_dict: dict, default = None
+            These input variables are optional and override the default settings.
+            Changing these variables likely required for more advanced systems.
+
+            The details of the acceptable input variables for the selected
+            ensembles can be found by running this python workbook,
+
+                print_valid_ensemble_input_variables('GCMC', description = True)
+
+            which prints the input_variables with their subsection description
+            for the selected 'GCMC' ensemble (other ensembles can be set as well).
+
+            Example : input_variables_dict = {'Restart' : False, 'PRNG' : 123,
+                                              'ParaTypeCHARMM' : True }
+
+        Returns
+        -------
+        None
+
+        Notes
+        -------
+        The details of the required inputs for the selected
+        ensembles can be found by running this python workbook,
+
+             print_valid_required_input_variables('NVT', description = True)
+
+        which prints the required inputs with their subsection description
+        for the selected 'NVT' ensemble (other ensembles can be set as well).
+
+        The box units imported are in nm.  They need and are to be converted to Ang for GOMC or NAMD
+        """
+
+        # set this to check and see if all the input pass
+        self.all_inputs_pass = True
+
+        # set this to check and see if all the input pass
+        self.all_failed_input_List = []
+
+        # Check if charmm_object is really a Charmm() object
+        if isinstance(charmm_object, mf_charmm.Charmm) == False:
+            self.all_inputs_pass = False
+            print_error_message = 'The variable supplied as a charmm_object ({}}) is not a ' \
+                                  'charmm_object ({}})'.format(type(mf_charmm.Charmm), type(mf_charmm.Charmm))
+            raise ValueError(print_error_message)
+
+        # check ensemble is a correct type
+        print('INFO: ensemble_type = ' + str(ensemble_type))
+        if ensemble_type in ['NPT', 'NVT', 'GCMC', 'GEMC_NVT', 'GEMC_NPT']:
+            self.ensemble_type = ensemble_type
+        else:
+            self.all_inputs_pass = False
+            print_error_message = "The ensemble_type is not a valid entry. "\
+                                  "Only 'NPT', 'NVT', 'GCMC', 'GEMC_NVT', 'GEMC_NPT' are valid entries."
+            raise ValueError(print_error_message)
```

Comment:
I dont think the `all_inputs_pass` attribute is actually being used here at all. Since you raise a `ValueError`, this exception will halt the program.

Reply:
If possible, I would like to keep this in there.  If the function fails and we need to detect it from a variable, not a failure, it will be present and readily avaialble.